Core/GDCore/Project/Project.cpp

```diff
@@ -618,6 +619,19 @@ void Project::UnserializeFrom(const SerializerElement& element) {
       propElement.GetStringAttribute("macExecutableFilename");
   useExternalSourceFiles =
       propElement.GetBoolAttribute("useExternalSourceFiles");
+
+  extensionProperties.UnserializeFrom(
+      propElement.GetChild("extensionProperties"));
+
+  // Compatibility: Move AdMob App ID from project property to extension
+  // property.
+  if (propElement.GetStringAttribute("adMobAppId", "") != "") {
+    extensionProperties.SetValue(
+        "AdMob",
+        "AdMobAppId",
+        propElement.GetStringAttribute("adMobAppId", ""));
+  }

```

Comment:
Use this exact format:
```
// Compatibility with GD <= 5.0.0-beta98
// Move AdMob App ID from project property to extension property.
...
// end of compatibility code
```

I try to consistent use these guards `Compatibility with GD <= ...` so that we can automate some code removal in the future :)